azurecontainerapps/azurecontainerapps.ts

```diff
@@ -111,15 +100,46 @@ export class azurecontainerapps {
                 }
             }
 
+            // Install the pack CLI if the Oryx builder is being used
+            if (shouldUseBuilder) {
+                await appHelper.installPackCliAsync();
+            }
+
+            // Set the Azure CLI to dynamically install missing extensions
+            util.setAzureCliDynamicInstall();
+
+            // Log in to Azure with the service connection provided
+            const connectedService: string = tl.getInput('connectedServiceNameARM', true);
+            authHelper.loginAzureRM(connectedService);
+
+            const acrUsername: string = tl.getInput('acrUsername', false);
+            const acrPassword: string = tl.getInput('acrPassword', false);
+
+            // Login to ACR if credentials were provided
+            // Note: this step should be skipped if we're ONLY using the YAML configuration file (no image to build/push/pull)
+            if (!shouldOnlyUseYaml && !util.isNullOrEmpty(acrName) && !util.isNullOrEmpty(acrUsername) && !util.isNullOrEmpty(acrPassword)) {
```

Comment:
As we have a couple of conditions checking for `!shouldOnlyUseYaml`, please consider creating one main `if` condition that checks for that. I expect that this would help the readability by basically saying "We're in the case where we need to build/push/pull an image here".

Reply:
Spending more time on the PR, it would be great if it was possible to divide the code into the couple of main cases we're tackling. Else it makes each `if` condition quite complex to follow.

Reply:
Agreed -- I thought about this when I added the conditional variables with this PR, but looking at the code on review, it could definitely be cleaned up to make this a lot clearer to navigate through. Let me refactor this.